src/pages/patientView/PatientViewPage.tsx

```diff
@@ -269,22 +265,36 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
                                             this.patientViewPageStore.sampleToDiscreteGenePanelId.result,
                                             this.patientViewPageStore.genePanelIdToEntrezGeneIds.result);
     }
-    
+
     @autobind
     @action toggleGenePanelModal(genePanelId?:string|undefined) {
         this.genePanelModal = {
             isOpen: !this.genePanelModal.isOpen,
             genePanelId: genePanelId || ''
         };
     }
-    
+
     @computed get modalSelectedGenePanel() {
         return this.patientViewPageStore.genePanelIdToPanel.result[this.genePanelModal.genePanelId];
     }
+    readonly sampleManager = remoteData({
+        await:()=>[
+            this.patientViewPageStore.patientViewData,
+            this.patientViewPageStore.clinicalEvents
+        ],
+        invoke:()=>{
+            const patientData = this.patientViewPageStore.patientViewData.result;
+            if (this.patientViewPageStore.clinicalEvents.result!.length > 0) {
+                return Promise.resolve(new SampleManager(patientData.samples!, this.patientViewPageStore.clinicalEvents.result));
+            } else {
+                return Promise.resolve(new SampleManager(patientData.samples!));
+            }
+        },
+        default: null
+    });
 
     public render() {
-
-        let sampleManager: SampleManager | null = null;
+        const sampleManager = this.sampleManager.result!;
```

Comment:
i know we've debated this before but i just find this so weird.  it's not a sampleManager, it's a sampleManagerPromise.   Data I can deal with being packaged with the business of its loading, but I a whole object like this is stranger.  I would rather just see a list of awaited promises out in the open.

Reply:
but it shouldnt be recomputed on every render. so then it would be in a @computed. but then we need to duplicate the list of dependencies in render() as in @computed which IMO is not good at all